### DISCUSSION TITLE
Samsung 2880 - Gentoo Works

### DISCUSSION THREAD

**Discussion by ericghara** [alexpevzner/sane-airscan#168] [Show and tell]
First of all Alex, thank you!  I spent an hour unsuccessfully trying to get this scanner to work with Samsung's unified drivers or hacking xerox_mfp. It only took me a few minutes to set-up with your driver! All features of the scanner work in wsd mode.  

Setup for gentoo-ers:

If you want a more compact sane-backends installation add to your /etc/portage/make.conf:
`SANE_BACKENDS="net"`

Make sure you emerge sane-backends with the zeroconf use flag enabled (allows avahi package to be installed).  If needed add this to your package.use:
`sane media-gfx/sane-backends zeroconf`

Then
`emerge -av media-gfx/sane-backends`

from there setup as in the readme.md
`git clone https://github.com/alexpevzner/sane-airscan.git`
`cd sane-airscan`
`make`
`make install`

Out of the box I was able to find my printer without touching the airscan.conf file.
`scanimage -L`
`device airscan:w0:Samsung Electronics Co., Ltd. M288x Series' is a WSD Samsung Electronics Co., Ltd. M288x Series ip=169.255.0.1, 192.168.1.64`

And scan from command line or with simple-scan

**Comment by ghost** [comment]
Cool!

**Comment by alexpevzner** [comment]
Thanks!

Your Gentoo-specific comments doesn't tell too much for me (I'm on Fedora), but may be helpful for other Gentoo users. If you want to share your experience, you may consider conrtibuting a Gentoo-specific section to the README.md

**Reply by jmvjmv57** [reply]
@alexpevzner 
Your airscan driver is now part of the Gentoo package tree under "media-gfx/sane-airscan". So a simple emerge (equivalent of yum or apt-get) will do the job. So no need for a Gentoo-specific section, but ericghara's information is useful.nontheless
